typed-protocols-examples/src/Network/TypedProtocol/Codec.hs

```diff
@@ -1,16 +1,16 @@
-{-# LANGUAGE GADTs                      #-}
-{-# LANGUAGE FlexibleContexts           #-}
-{-# LANGUAGE TypeFamilies               #-}
-{-# LANGUAGE RankNTypes                 #-}
-{-# LANGUAGE PolyKinds                  #-}
-{-# LANGUAGE DataKinds                  #-}
-{-# LANGUAGE ScopedTypeVariables        #-}
-{-# LANGUAGE NamedFieldPuns             #-}
-{-# LANGUAGE TypeInType                 #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeInType            #-}
 -- @UndecidableInstances@ extension is required for defining @Show@ instance of
 -- @'AnyMessage'@.
```

Comment:
This comment should apply to `UndecidableInstances`

Reply:
And if we are this could you update it, it mentioned `'AnyMessage'` while it should `AnyMessageAndAgency'.

Reply:
I looks like it is needed for both:

```
src/Network/TypedProtocol/Codec.hs:288:10: error:
    • The constraint ‘Show (Message ps st st')’
        is no smaller than the instance head ‘Show (AnyMessage ps)’
      (Use UndecidableInstances to permit this)
    • In the instance declaration for ‘Show (AnyMessage ps)’
    |
288 | instance (forall st st'. Show (Message ps st st')) => Show (AnyMessage ps) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Network/TypedProtocol/Codec.hs:303:5: error:
    • The constraint ‘Show (ClientHasAgency st)’
        is no smaller than
        the instance head ‘Show (AnyMessageAndAgency ps)’
      (Use UndecidableInstances to permit this)
    • In the instance declaration for ‘Show (AnyMessageAndAgency ps)’
    |
303 |     ( forall (st :: ps)            . Show (ClientHasAgency st)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```